File: src/17.ts
import fs from 'fs'

const source = fs.readFileSync('./res/popular-names.txt', 'utf-8')
console.log(
    [...new Set(
        source
            .split('\n')
            .filter(row => row.length > 0)
            .map(row => row[0])
    )].sort()
)

File: src/17.py
# cut -c 1 res/popular-names.txt | sort | uniq
from operator import itemgetter

from .helper.file_helper import read_string

source = read_string("./res/popular-names.txt")
print("\n".join(sorted(set(map(itemgetter(0), source.split("\n"))))))
